semeio/jobs/rft/gendata_rft.py

```diff
@@ -1,27 +1,86 @@
 import logging
+import pandas as pd
 
 logger = logging.getLogger(__name__)
 
 
-def _write_points(points, fname):
-    with open(fname, "w+") as fh:
-        fh.write("\n".join([str(p.get_pressure()) for p in points]))
+def _write_active_pressure_points(trajectory_df, fname):
+    """Generate three files with the information GENDATA needs
+    from the trajectory dataframe.
+
+    Args:
+        trajectory_df (pd.DataFrame): The column "order" should contain
+            the original row order from input text files. "pressure" should
+            hold pressure data, but will be defaulted if not present. "is_active"
+            should be a boolean column, and "inactive_info" should contain
+            strings with information on why points are inactive.
+    """
+    # Write pressure value, one pr line for all points, -1 is used where
+    # there is no pressure information.
+    with open(fname + "", "w+") as fh:
+        if "pressure" in trajectory_df:
+            fh.write(
+                "\n".join(
+                    trajectory_df.sort_values("order")["pressure"]
+                    .fillna(value=-1)
+                    .astype(str)
+                    .values
+                )
+                + "\n"
+            )
+        else:
+            fh.write("\n".join(["-1" * len(trajectory_df)]))
+
     logger.info("Forward model script gendata_rft.py: Wrote file {}".format(fname))
 
+    # Write a file with "1" pr row if a point is active, "0" if not
     with open(fname + "_active", "w+") as fh:
-        fh.write("\n".join([str(int(p.is_active())) for p in points]))
-
-
-def _write_inactive_info(points, fname, zonemap=None):
```

Comment:
Is `zonemap` no longer needed here? Is it just that it is handled elsewhere?

Reply:
It is now handled on dataframe construction.